shell-ui/src/NavBar.js

```diff
@@ -2,64 +2,101 @@
 import CoreUINavbar from '@scality/core-ui/dist/components/navbar/Navbar.component';
 import { useAuth } from 'oidc-react';
 import { useLayoutEffect, useState } from 'react';
-import type { SolutionsNavbarProps } from './index';
+import type {
+  Options,
+  SolutionsNavbarProps,
+  TranslationAndGroups,
+} from './index';
 import type { Node } from 'react';
 import { logOut } from './auth/logout';
+import {
+  getAccessiblePathsFromOptions,
+  isEntryAccessibleByTheUser,
+  normalizePath,
+} from './auth/permissionUtils';
+import { prefetch } from 'quicklink';
 
-export const LoadingNavbar = (): Node => <CoreUINavbar role="navigation" tabs={[{title: 'loading'}]} />;
+export const LoadingNavbar = (): Node => (
+  <CoreUINavbar role="navigation" tabs={[{ title: 'loading' }]} />
+);
 
-export const Navbar = ({
-  options,
-}: {
-  options: $PropertyType<SolutionsNavbarProps, 'options'>,
-}): Node => {
+const translateOptionsToMenu = (
+  options: Options,
+  section: 'main' | 'subLogin',
+  renderer: (
+    path: string,
+    translationAndGroup: TranslationAndGroups,
+  ) => { link: Node } | { label: string, onClick: () => void },
+  userGroups: string[],
+) => {
+  const normalizedLocation = normalizePath(location.href);
+  return (
+    Object.entries(options[section])
+      //$FlowIssue - flow typing for Object.entries incorrectly typing values as [string, mixed] instead of [string, TranslationAndGroups]
+      .filter((entry: [string, TranslationAndGroups]) =>
+        isEntryAccessibleByTheUser(entry, userGroups),
+      )
+      .map(
+        //$FlowIssue - flow typing for Object.entries incorrectly typing values as [string, mixed] instead of [string, TranslationAndGroups]
+        ([path, translationAndGroup]: [string, TranslationAndGroups], i) => {
+          try {
+            return {
+              ...renderer(path, translationAndGroup),
+              selected: translationAndGroup.activeIfMatches
+                ? new RegExp(translationAndGroup.activeIfMatches).test(
+                    location.href,
+                  )
+                : normalizedLocation === normalizePath(path),
```

Comment:
I also thought about switching to startsWith but IMO it is not a correct approach because we can for example have a menu entry for `/` and for `/alerts` on the same UI where this default will make both entries selected. I think an exact match is the strongest default to avoid duplicated selected entries that could mislead the user.